src/test/java/org/elasticsearch/client/AbstractClientHeadersTests.java

```diff
@@ -84,16 +85,19 @@
                 CreateIndexAction.INSTANCE, IndicesStatsAction.INSTANCE, ClearIndicesCacheAction.INSTANCE, FlushAction.INSTANCE
     };
 
+    protected ThreadPool threadPool;
     private Client client;
 
     @Before
     public void initClient() {
+        threadPool = new ThreadPool("test");
```

Comment:
give it the tests name that might make it easier to figure out where it leaked if it leaked